client/src/views/page-components/account/AccountHistoryModal/AccountHistoryModal.ts

```diff
@@ -0,0 +1,388 @@
+    const historyContent: HTMLInputElement = document.querySelector('.history-form__content');
+
+    if (
+      this.payMethod.currentCardName.length > 0 &&
+      this.dateValueValidation &&
+      this.moneyValueValidation &&
+      this.choicedCategoryName.length > 0 &&
+      historyContent.value.length > 0
+    ) {
+      this.checkIncomeOrExpenditure();
+
+      const submitArguments = {
+        //   user:,
+        card: this.payMethod.currentCardName,
+        category: this.choicedCategoryName,
+        money: this.moneyInput.value as HTMLInputElement,
+        date: this.dateInput.value as HTMLInputElement,
+        content: historyContent.value,
+        type: this.type,
+      };
+      console.log('Form Success');
+
+      this.closeModal();
+      handleEvent.fire('createaccounthistory', { state: history.state, submitArguments });
+      // 옵저버 발동
+    } else {
+      this.showAlert('.history-form__confirm-alert');
+    }
+  }
+
+  checkIncomeOrExpenditure() {
+    if (this.choicedCategoryName === '용돈') {
+      this.type = 'income';
+    } else {
+      this.type = 'expenditure';
+      this.moneyInput.value = '-' + this.moneyInput.value;
+    }
+  }
+  onKeyUpHandler(e: KeyboardEvent) {
+    this.onKeyUpDate(e);
+    this.onKeyUpMoney(e);
+  }
+
+  onFocusOutDateInputHandler() {
+    this.onFocusOutDate();
+  }
+
+  onFocusOutMoneyInputHandler() {
+    this.onFocusOutMoney();
+  }
+
+  onFocusOutDate() {
+    const target = $('.history-form__date');
+    const ValidationLengthResult = this.checkDateInputLengthValidation(target, dateInputMaxLength);
+    const ValidationValueResult = this.checkDateInputValueValidation(target);
+
+    if (ValidationLengthResult || ValidationValueResult) {
+      this.showAlert('.date-alert');
+      this.dateValueValidation = false;
+    } else {
+      this.dateValueValidation = true;
+    }
+    this.formatDateValue();
+  }
+
+  formatDateValue() {
+    const DateValue: string = this.dateInput.value;
+    const DateArray = DateValue.split('');
+    // 4, 7
+    if (DateValue.length > 6) {
+      DateArray.splice(4, 0, '.');
+      DateArray.splice(7, 0, '.');
+      this.dateInput.value = DateArray.join('');
+    } else if (DateValue.length > 4) {
+      DateArray.splice(4, 0, '.');
+      this.dateInput.value = DateArray.join('');
+    }
+  }
+
+  onFocusOutMoney() {
+    const target = $('.history-form__money');
+    const ValidationResult = this.checkMoneyInputLengthValidation(target, moneyInputMinLength);
+
+    if (ValidationResult) {
+      this.showAlert('.money-alert');
+      this.moneyValueValidation = false;
+    } else {
+      this.moneyValueValidation = true;
+    }
+    this.formatMoneyValue();
+  }
+
+  formatMoneyValue() {
+    const MoneyValue: string = this.moneyInput.value;
+    const MoneyArray = MoneyValue.split('');
+    const MoneyLength = MoneyValue.length;
+
+    if (MoneyLength > 3) {
+      let count = 1;
+      for (let i = MoneyLength - 1; i > 0; i--) {
```

Comment:
우리 변수는 camelCase 쓰기로 했잔앋여~~~
```suggestion
    const moneyValue: string = this.moneyInput.value;
    const moneyArray = moneyValue.split('');
    const moneyLength = moneyValue.length;

    if (moneyLength > 3) {
      let count = 1;
      for (let i = MoneyLength - 1; i > 0; i--) {
```

Reply:
피곤해서 실수요 ..ㅎㅎ